wikidataQuery.py

```diff
@@ -1,16 +1,50 @@
 # Functions needed for making queries for WikiData
 import requests
 
+# A pre-defined dictionary for difficult terms
+property_dict = {'band members': 'has part', 'members': 'has part',
+                  'member': 'has part', 'band member': 'has part',
+                  'founding year': 'inception', 'bandmember': 'has part',
+                  'bandmembers': 'has part', 'founding': 'inception',
+                  'play': 'instrument'}
+
+# List of w-words, feel free to add any words I forgot
+w_words_list = ['What', 'Who', 'When', 'Where', 'Why', 'How', 'Which']
+
 def makeQuery(keywords):
     property_id = None
     entity_id = None
+    prop_attribute_id = None
+    
+    query_type = None
+
     for keyword in keywords:
-        if keyword[1] == "property":
-            property_id = searchEntity(keyword[0], "property")
+        if keyword[1] == "question_word":
+            if keyword[0] in w_words_list:
+                #TODO update property & possibly query based on that
+                query_type = 'basic'
+            else:
+                query_type = 'yes/no' 
+                
+        elif keyword[1] == "property":
+            prop = property_dict.get(keyword[0], keyword[0])
+            property_id = searchEntity(prop, "property")
+            
         elif keyword[1] == "entity":
             entity_id = searchEntity(keyword[0], "entity")
+            
+        elif keyword[1] == "property_attribute":
+            # TODO attribute is not always entity, right? needs to be fixed
+            prop_attribute_id = searchEntity(keyword[0], "entity")
+            
+    answer = None
```

Comment:
Alternatively, can't you simply use `return submitQuery()` and `return submitCheckQuery()`? I think that works out fine in python, even if neither if-statement is entered.

Reply:
I prefer one return, it's much more elegant

Reply:
I prefer multiple returns, it's more elegant. ;)